blog/content/post/2020-02-18-ew2020-btg-demo.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Boot Guard - pre-execution firmware verification
+abstract: 'This post will not describe how to guard your shoes. However,
+          will definitely introduce you to Boot Guard feature present on
+          Intel processors which allows firmware verification before the first
+          instruction executes. One may call it pre-execution firmware
+          verification.'
+cover: /covers/boot_guard.png
+author: michal.zygowski
+layout: post
+published: true
+date: 2020-02-18
+archives: "2020"
+
+tags:
+  - Boot Guard
+  - Secure Boot
+  - coreboot
+  - Protectli
+categories:
+  - Firmware
+  - Security
+
+
+---
+### What Boot Guard is?
+
+For sure, Boot Guard does not mean exactly what its name suggest, it is not
+protecting shoes. In firmware and BIOS nomenclature `boot` means the bootstrap
+process, which is the early processor initialization routine. Almost every
+modern machine or processor has some kind of firmware which is responsible for
+the bootstrap process - booting - which prepares the silicon to load the target
+application or operating system. So Boot Guard means that it protects the boot
+process and the firmware. But from what should it protect and why? You can find
+it out by spending few spare minutes and reading this post.
+
+![Boot + Guard](/covers/boot_guard.png)
+
+---
+### Why Boot Guard?
+
+Nowadays security researchers are very focused on trustful solutions, security
+technologies and low-level safety of the computer machines used everyday. They
+explore every possibility and attack vector to find vulnerabilities in the most
+crucial parts of security in modern systems, like: Intel ME, UEFI etc. By
+exposing discovered threats the silicon vendors release fixes to protect the
+hardware and software from being compromised.
+
+How do you know that software you use is trustful, secure and hasn't been
+maliciously modified before execution? One can only ensure that by establishing
+the Root of Trust (ROT) as early as possible. There are 2 main type of ROTs:
+
+- SRTM - Static Root of Trust for Measurement
+- DRTM - Dynamic Root of Trust for Measurement
+
+I will not explain the measurement process as it is not a part of this post.
+What I will focus on is cryptographical verification in the ROT. ROT is the
+beginning of the verification process, it is considered to be trusted and
+verifies all components executed by firmware.
```

Comment:
@miczyg1 how this connects with mentioned acronyms? Can you point to DRTM explanation or other TrenchBoot blog/conference paper?

Reply:
Added links few line below. Something like that will be ok?

Reply:
@miczyg1 ok, but you should probably directly say that Intel Boot Guard is SRTM implementation.

Reply:
Done